core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -240,49 +241,53 @@ public boolean isUsingSecurityToken() {
      * Called during the initial setup to create an admin user
      */
     @RequirePOST
-    public HttpResponse doCreateAdminUser(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
+    public HttpResponse doCreateAdminUser(StaplerRequest req, StaplerResponse rsp) throws IOException {
```

Comment:
Should be ideally restricted

Reply:
Sorry, I don't understand what you mean. From what I see, permission is checked in the second line with `checkPermission`. 
If you're referring to the `IOException`, that very class is actually thrown from the method body.

Reply:
I meant adding the `@Restricted(NoExternalUse.class)` annotation so that we are sure that nobody tries to use this method externally.

Generally your change is binary compatible, so it is fine anyway

Reply:
Oh, yeah, that sounds reasonable. It's half past 1 am here, so I'll add that tomorrow – or today, depending on how you see it.

Reply:
"Today" for me as well :)
It is not a blocker for the merge tho